### PR TITLE
Add size calculation method and trigger mechanism

### PR DESCRIPTION
**Pull request by loloxwg**
Implemented a `size` method for `Record` and added a trigger mechanism (`Trigger`, `TriggerFactory`) to handle dynamic item insertion and manage memory efficiently. Adjusted related stream implementations and updated dependency path for `tonbo_marco`.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -49,7 +49,7 @@ regex = "1"
 thiserror = "1"
 tokio = { version = "1", features = ["fs", "io-util"] }
 tokio-util = { version = "0.7" }
-tonbo_marco = "0.0.1"
+tonbo_marco = {path="tonbo_marco"}
```
Comment (Xuanwo) [inline Cargo.toml:52]: This could prevent us from making a release. Please also specify a version.

**Review by ethe — APPROVED** [review]

File: src/inmem/mutable.rs [hunk 2]
```diff
@@ -55,9 +57,15 @@ where
             wal = Some(Mutex::new(WalFile::new(file, file_id)));
         };
 
+        let trigger = Arc::new(TriggerFactory::create(
```
Comment (KKould) [inline src/inmem/mutable.rs:60]: currently, using ``option.max_mutable_len when constructing a trigger can cause confusion. I think using an enumeration like 
```rust
enum TriggerType {
    SizeOfMem(usize),
    Length(usize),
}
```
then `TriggerFactory::create` uses this enum as parameter

File: src/option.rs [hunk 3]
```diff
@@ -37,10 +39,10 @@ where
     /// build the default configured [`DbOption`](struct.DbOption.html) based on the passed path
     fn from(path: P) -> Self {
         let (column_paths, sorting_columns) = R::primary_key_path();
-
+        let max_mutable_len = 3000;
         DbOption {
             path: path.into(),
-            max_mutable_len: 3000,
+            max_mutable_len,
```
Comment (KKould) [inline src/option.rs:45]: `max_mutable_len` is already included in `trigger_type`, so it can be removed

File: src/compaction/mod.rs [hunk 4]
```diff
@@ -464,7 +472,9 @@ pub(crate) mod tests {
         R: Record + Send,
         FP: FileProvider,
     {
-        let mutable: Mutable<R, FP> = Mutable::new(option).await?;
+        let trigger = Arc::new(TriggerFactory::create(option.trigger_type));
+
+        let mutable: Mutable<R, FP> = Mutable::new(option, trigger.clone()).await?;
```
Comment (KKould) [inline src/compaction/mod.rs:477]: remove useless `clone`

**Review by KKould — APPROVED** [review]
LGTM

**Comment by ethe** [comment]
Thank you @loloxwg ! I hope there are more collaborations in the future.

**Pull request merged**